pyanaconda/modules/network/nm_client.py

```diff
@@ -905,3 +904,32 @@ def finish_callback(connection, result, sync_queue):
     )
 
     return sync_queue.get()
+
+
+def activate_connection_sync(nm_client, connection, device):
+    """Activate a connection synchronously.
+
+    Synchronous wrapper of ActivateConnection() NM method.
+
+    :param connection: NetworkManager connection
+    :type connection: NM.RemoteConnection
+    :param device: the preferred device to apply the connection to
+                   None if not needed
+    :type device: NM.Device
+    """
+    sync_queue = Queue()
+
+    def finish_callback(nm_client, result, sync_queue):
+        ret = nm_client.activate_connection_finish(result)
+        sync_queue.put(ret)
+
+    nm_client.activate_connection_async(
+        connection,
+        device,
+        None,
+        None,
+        finish_callback,
+        sync_queue
+    )
+
+    return sync_queue.get()
```

Comment:
You already have a pretty similar function on https://github.com/rhinstaller/anaconda/pull/2092/files#diff-bf59496c5abcb42d684f27494eb56ef6R871 maybe it would be great to create a separate helper function to handle the callback solution?

Reply:
I don't see any useful generalization/abstraction possible here. Apart from using queue as callback synchronisation I don't see anything common in the functions.

Reply:
Hmm, I see there are too many differences even when it looks the same on the first look. OK, you can ignore this.